source/DataBase/DataBaseConnection.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <string>
+#include <stdexcept>
+
+#include "interface/IDataBaseConnection.hpp"
+#include "sqlite3.h"
+
+namespace dataBase
+{
+    class DataBaseConnection : public interface::IDataBaseConnection
+    {
+    public:
+        DataBaseConnection(const char *dbName);
+
+        ~DataBaseConnection();
+
+        int executeQuery(const std::string &query, sqlite3_callback selectCallback = nullptr, std::shared_ptr<common::ITable> records = nullptr) override;
+
+    private:
+        sqlite3 *database_;
```

Comment:
unique_ptr

Reply:
It is not possible with sqlite3. Raw pointer is required here